test/groovy/com/sap/piper/MtaUtilsTest.groovy

```diff
@@ -0,0 +1,247 @@
+package com.sap.piper
+
+import org.junit.Before
+import org.junit.ClassRule
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.ExpectedException
+import org.junit.rules.RuleChain
+import org.junit.rules.TemporaryFolder
+import org.yaml.snakeyaml.Yaml
+
+import com.lesfurets.jenkins.unit.BasePipelineTest
+
+import groovy.json.JsonSlurper
+import hudson.AbortException
+import util.JenkinsEnvironmentRule
+import util.Rules
+
+
+class MtaUtilsTest extends BasePipelineTest {
+    private static srcPackageJson = 'test/resources/MtaUtils/package.json'
+    private static mtaTemplate = 'resources/template_mta.yml'
+    private static data
+    private static targetMtaDescriptor
+    private File badJson
+    private mtaUtils
+
+    @Rule
+    public JenkinsEnvironmentRule jer = new JenkinsEnvironmentRule(this)
+
+
+    @Rule
+    public ExpectedException thrown= ExpectedException.none();
+
+    @ClassRule
+    public static TemporaryFolder tmp = new TemporaryFolder()
+
+    @Rule
+    public RuleChain ruleChain = Rules
+    .getCommonRules(this)
+    .around(jer)
+
+    void prepareObjectInterceptors(object) {
+        object.metaClass.invokeMethod = helper.getMethodInterceptor()
+        object.metaClass.static.invokeMethod = helper.getMethodInterceptor()
+        object.metaClass.methodMissing = helper.getMethodMissingInterceptor()
+    }
+
+    @Before
+    void init() {
+        targetMtaDescriptor = "${tmp.getRoot()}/generated_mta.yml"
+        def script = new Object()
+        mtaUtils = new MtaUtils(script)
+
+        prepareObjectInterceptors(script)
+
+        this.helper.registerAllowedMethod('readJSON', [Map], { Map parameters ->
+            def js = new JsonSlurper()
+            def reader = new BufferedReader(new FileReader(parameters.file))
+            def j = js.parse(reader)
+            return j
+        })
+
+        this.helper.registerAllowedMethod('libraryResource', [Map], {  Map parameters ->
+            new Yaml().load(new File(mtaTemplate).text)
+        })
+
+        this.helper.registerAllowedMethod('readYaml', [], {
+            return new Yaml().load(new FileReader(mtaTemplate))
+        })
+
+        this.helper.registerAllowedMethod('writeYaml', [Map], { Map parameters ->
+            def generatedFile = new File(parameters.file)
+            generatedFile.text = parameters.data
+            generatedFile.dump()
```

Comment:
Dumping the file here and asserting that this file exists later on in test ```testFileGenerated``` asserts in fact only that the closure works. This does not add any values, especially it does not assert that the file exist later on when the real writeYaml coding is called. And it is not our duty to check ```writeYaml``` here. This is (hopefully) covered by tests inside the plugin providing ```writeYaml```.

It is IMO sufficient to remember the ```data``` here in order to be able to check the details later in a test (as it is done in ```straightForwardTest```. It would be also a benefit to remember the file name in order to be able to apply also a check for the file name later in a test.

Since dumping the file does not provide any value this should be removed.

Reply:
done